auth/token.go

```diff
@@ -30,6 +37,70 @@ const (
 	UsernameClaimKey = "username"
 )
 
+func init() {
+	// ensures we don't generate predictable token signing keys
+	rand.Seed(time.Now().UnixNano())
+}
+
+var tokenCharacters = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890")
+
+func generateTokenSigningKey() string {
```

Comment:
I don't think there's any value in extracting this into a reusable function right now.  The valid characters are also specific to the token signing key.

This function is being called extremely infrequently.  The cumulative CPU time of everyone who will ever have this code execute on their systems would never add up to the time it would take to replace it with an optimized version.  The third option there also doesn't have the same probability for all characters to be chosen.